Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -433,12 +433,19 @@ WebIDL::ExceptionOr<void> XMLHttpRequest::send(Optional<Fetch::XMLHttpRequestBod
     auto request = LoadRequest::create_for_url_on_page(request_url, m_window->page());
     request.set_method(m_method);
     if (body_with_type.has_value()) {
-        TRY_OR_RETURN_OOM(realm, body_with_type->body.source().visit([&](ByteBuffer const& buffer) -> ErrorOr<void> {
+        TRY(body_with_type->body.source().visit(
+            [&](ByteBuffer const& buffer) -> WebIDL::ExceptionOr<void> {
                 request.set_body(buffer);
```

Comment:
Isn't this also fallible? ByteBuffer has a silly copy ctor, hiding all the implicit copies of potentially huge buffers...

Reply:
Would it help if there was a `try_` implementation of the `set_body()` method. That's the only way I can think of at the moment that can can handle the fallible case.
```cpp
ErrorOr<void> LoadRequest::try_set_body(ByteBuffer const& body) { m_body = TRY(ByteBuffer::copy(body.bytes())); }
```

Reply:
Let's try this